client/discovery/discovery.js

```diff
@@ -0,0 +1,187 @@
+window.onload = function() {
+  if(typeof switch_is_present == 'undefined') mountDiscovery()
+}
+
+function mountDiscovery() {
+    "use strict"
+
+    var width,height
+    var chartWidth, chartHeight
+    var margin
+    var svg = d3v4.select("#graph").append("svg")
+    var chartLayer = svg.append("g").classed("chartLayer", true)
+
+    main()
+
+    function main() {
+        var range = 100
+        var data = {
+            // nodes:d3v4.range(0, range).map(function(d){ return {label: "l"+d ,r:~~d3v4.randomUniform(8, 28)()}}),
+            nodes: d3v4.range(0,range).map( (d) => { return {label: "l"+d , r: 10 }}),
+            links:d3v4.range(0, range).map(function(){ return {source:~~d3v4.randomUniform(range)(), target:~~d3v4.randomUniform(range)()} })
+        }
+
+        setSize(data)
+        drawChart(data)
+    }
+
+    function setSize(data) {
+        width = document.querySelector("#graph").clientWidth
+        height = document.querySelector("#graph").clientHeight
+        margin = {top:0, left:0, bottom:0, right:0 }
+
+
+        chartWidth = width - (margin.left+margin.right)
+        chartHeight = height - (margin.top+margin.bottom)
+
+        svg.attr("width", width).attr("height", height)
+
+
+        chartLayer
+            .attr("width", chartWidth)
+            .attr("height", chartHeight)
+            .attr("transform", "translate("+[margin.left, margin.top]+")")
+
+
+    }
+
+    function drawChart(data) {
+
+        var simulation = d3v4.forceSimulation()
+            .force("link", d3v4.forceLink().id(function(d) { return d.index }))
```

Comment:
arrow function

Reply:
and below too

Reply:
why? to make sure it will not run in all browsers?

Reply:
we're not using ES2015 here